File: src/snippets/rate-limiting/reference/claude-managed-agents/js/DecisionLog.ts
import { launchArcjet, detectPromptInjection } from "@arcjet/guard";

const arcjet = launchArcjet({ key: process.env.ARCJET_KEY! });
const inbound = detectPromptInjection();

export async function screenPrompt(prompt: string) {
  const decision = await arcjet.guard({
    label: "message.received",
    rules: [inbound(prompt)],
  });

  console.log("conclusion", decision.conclusion);
  console.log("reason", decision.reason);
  console.log("failed open", decision.hasFailedOpen());

  for (const result of decision.results) {
    console.log("rule result", result);
  }
}

File: src/snippets/rate-limiting/reference/langchain/py/DecisionLog.py
import os

from arcjet.guard import DetectPromptInjection, launch_arcjet

arcjet = launch_arcjet(key=os.environ["ARCJET_KEY"])
inbound = DetectPromptInjection()


async def screen_prompt(prompt: str) -> None:
    decision = await arcjet.guard(
        label="message.received",
        rules=[inbound(prompt)],
    )
    print("conclusion", decision.conclusion)
    print("reason", decision.reason)
    print("failed open", decision.has_failed_open())
    for result in decision.results:
        print("rule result", result)
